src/Rector/Deprecation/WatchdogExceptionRector.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace DrupalRector\Rector\Deprecation;
+
+use DrupalRector\Contract\VersionedConfigurationInterface;
+use DrupalRector\Rector\AbstractDrupalCoreRector;
+use DrupalRector\Rector\ValueObject\DrupalIntroducedVersionConfiguration;
+use PhpParser\Node;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+class WatchdogExceptionRector extends AbstractDrupalCoreRector
+{
+    /**
+     * @var array|DrupalIntroducedVersionConfiguration[]
+     */
+    protected array $configuration;
+
+    /**
+     * @inheritdoc
+     */
+    public function getNodeTypes(): array {
+        return [
+            Node\Expr\FuncCall::class
+        ];
+    }
+
+    public function configure(array $configuration): void {
+        foreach ($configuration as $value) {
+            if (!($value instanceof DrupalIntroducedVersionConfiguration)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Each configuration item must be an instance of "%s"',
+                    DrupalIntroducedVersionConfiguration::class
+                ));
+            }
+        }
```

Comment:
We should create an assert helper. Drupal core's `Inspector::assertAll`: https://git.drupalcode.org/project/drupal/-/blob/11.x/core/lib/Drupal/Component/Assertion/Inspector.php#L53.

Maybe we could add a `require` on the following (or just copy relevant parts

```
"drupal/core-assertion": "^8 || ^9 || ^10 || ^11
```

Yeah.. let's just copy it into a new PR and simplify `configure`

Reply:
Good idea :)